src/code_memory/server.py

```
@@ -0,0 +1,199 @@
+        return f"Error: {str(e)}"
+    except Exception as e:
+        ctx.error(f"Unexpected error: {str(e)}")
+        logger.error("Failed to store code snippet", exc_info=True)
+        return "An unexpected error occurred"
+
+
+@mcp.tool()
+async def search_code(
+    query: str,
+    ctx: Context,
+    language: str = None,
```

Ruff (0.11.9)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)